### PR TITLE
JBPM-5381: Stunner & Guvnor & WB commons integration.

### PR DESCRIPTION
**Pull request by romartin**
Hey @manstis , @tsurdilo , @jlindop !
Here is the Stunner integration with KIE workbench & guvnor commons screens, editors and services.

See -> https://issues.jboss.org/browse/JBPM-5381
Depends on -> https://github.com/droolsjbpm/droolsjbpm-build-bootstrap/pull/302

Some updates:
- Refactors, improvements and fixings on backend side, which was really still a "proptotype".
- Added all workbench & guvnor integration stuff into children modules of kie-wb-common-stunner-project
- Added the concrete resource type, handlers and editors for BPMN - module kie-wb-common-bpmn-project
- Added another showcase that shows the integration stuff - module kie-wb-common-stunner-showcase-project
- Current existing showcase still present, as it loads double faster and consumes half of memory compared to the project showcase, and renamed to kie-wb-common-stunner-project-standalone
- Removed kie-wb-common-stunner-workbench module, the screens presents on that module has been moved to the standalone showcase.
- Added some tests on different modules
- Refactored the BPMN file suffix to "bpmn2" everywhere as the current jbpm-designer does. Refactored suffix on all files on the project as well.

Still improving and adding new tests and doing bug fixing, but the integration and showcases I are ready enough for merging and letting others to continue with the work (depends on this). Thanks!

### CODE REVIEW

**Comment by romartin** [comment]
oopps, trying to reproduce and fix locally.

**Comment by romartin** [comment]
Added new commit that fix the Jenkins build, sorry it was missing a source artifact. Rebuilding

**Comment by romartin** [comment]
Green again :)

**Comment by tsurdilo** [comment]
@romartin this looks good to me. I can run the tests from your branch. 

So +1 just one question - 
I don't see where the association to a asset type definitions is (or will it be added (for example  BpmnTypeDefinition implements ResourceTypeDefinition { ... }) - thanks!

**Comment by romartin** [comment]
Hey @tsurdilo ! Good! Btw the asset definition is inside the bpmn modules - >https://github.com/romartin/kie-wb-common/tree/stunner-project-integration/kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-project

**Comment by romartin** [comment]
Hey guys, as this ticket has not been merged yet, just added right now a new commit for fixing an issue.  Jeknins should be building again. Thanks!

**Comment by romartin** [comment]
Rebased with master again to avoid Jenkins complains. 
Jenkins please retest this.

**Comment by romartin** [comment]
Hi, as not merged yet, I have added some minor fixes and renamed again the use of bpmn2 extensions for Stunner to just "bpmn", doing so allows the use of both process designers on the kie-wb as they're become different resource types. 
Rebuilding... Thanks!

**Comment by romartin** [comment]
Hi @csadilek , just realized that @manstis is on PTO this week.. please could you take a look at this one?

File: kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/AbstractDefinitionSetServices.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+package org.kie.workbench.common.stunner.backend.service;
+
+import org.kie.workbench.common.stunner.core.definition.adapter.binding.BindableAdapterUtils;
+import org.kie.workbench.common.stunner.core.definition.service.DefinitionSetServices;
+import org.kie.workbench.common.stunner.core.definition.service.DiagramMarshaller;
+import org.kie.workbench.common.stunner.core.diagram.Diagram;
+import org.kie.workbench.common.stunner.core.diagram.Metadata;
+import org.kie.workbench.common.stunner.core.graph.Graph;
+
+public abstract class AbstractDefinitionSetServices implements DefinitionSetServices {
```
Comment (csadilek) [inline kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/AbstractDefinitionSetServices.java]: DefinitionSetService is a better name, why plural?

File: kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/AbstractDefinitionSetServices.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    private final DiagramMarshaller<Graph, Metadata, Diagram<Graph, Metadata>> diagramMarshaller;
+
+    protected AbstractDefinitionSetServices() {
+        this( null );
```
Comment (csadilek) [inline kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/AbstractDefinitionSetServices.java]: Consider using Java 8 Optional

File: kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/AbstractDiagramLookupService.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    protected org.uberfire.java.nio.file.Path parseCriteriaPath( DiagramLookupRequest request ) {
+        String criteria = request.getCriteria();
+        if ( isEmpty( criteria ) ) {
+            LOG.error( "Empty criteria not supported." );
+            throw new UnsupportedOperationException( "Empty criteria not supported." );
+        } else {
+            Map<String, String> criteriaMap = AbstractCriteriaLookupManager.parseCriteria( criteria );
+            String pathRaw = criteriaMap.get( "path" );
+            if ( !isEmpty( pathRaw ) ) {
+                // TODO
```
Comment (csadilek) [inline kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/AbstractDiagramLookupService.java]: TODO what?

File: kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/AbstractDiagramLookupService.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                throw new UnsupportedOperationException( "No path criteria found." );
+            }
+        }
+    }
+
+    protected BaseDiagramService<D> getDiagramService() {
+        return diagramService;
+    }
+
+    protected boolean isEmpty( String s ) {
```
Comment (csadilek) [inline kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/AbstractDiagramLookupService.java]: not needed StringUtils are on the classpath.

File: kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/DiagramLookupServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+                    }
+
+                }
+                LOG.error( "Diagram with name [" + name + "] not found." );
+                return null;
+            }
+        }
+        String m = "Criteria [" + criteria + "] not supported.";
+        LOG.error( m );
+        throw new UnsupportedOperationException( m );
```
Comment (csadilek) [inline kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/DiagramLookupServiceImpl.java]: That's not the right exception type, as clearly the operation is supported.

File: kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/DiagramServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+    private static final String METADATA_EXTENSION = "meta";
+    private static final String VFS_ROOT_PATH = "default://stunner";
+    private static final String VFS_DIAGRAMS_PATH = "diagrams";
+    private static final String APP_DIAGRAMS_PATH = "WEB-INF/diagrams";
+
+    private FileSystem fileSystem;
+    private org.uberfire.java.nio.file.Path root;
+
+    protected DiagramServiceImpl() {
+        this( null, null, null, null, null, null );
```
Comment (csadilek) [inline kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/DiagramServiceImpl.java]: eeeek is this for testing purposes?

File: kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/DiagramServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+        // Initialize the application's VFS.
+        initFileSystem();
+
+        // Register packaged diagrams into VFS.
+        registerAppDefinitions();
+
+        // Load vfs diagrams and put into the parent registry.
+        final Collection<Diagram> diagrams = getAllDiagrams();
+        if ( null != diagrams && !diagrams.isEmpty() ) {
+            for ( Diagram diagram : diagrams ) {
```
Comment (csadilek) [inline kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/DiagramServiceImpl.java]: start using Java8. diagrams.forEach...

File: kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/DiagramServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,289 @@
...
+
+        org.uberfire.java.nio.file.Path path = null;
+
+        try {
+
+            path = getDiagramsPath().resolve( getMetadataFileName( dPath.getFileName() ) );
+
+        } catch ( org.uberfire.java.nio.file.NoSuchFileException e ) {
+
+            return null;
```
Comment (csadilek) [inline kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/DiagramServiceImpl.java]: seems worth throwing an exception if the file isn't found, no?

File: kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/DiagramServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,289 @@
...
+            File[] files = new File( directory ).listFiles( _deployFilter );
+            if ( files != null ) {
+                for ( File f : files ) {
+                    try {
+                        String name = f.getName();
+                        if ( isFileNameAccepted( name ) ) {
+                            // Register it into VFS storage.
+                            registerIntoVFS( f );
+                            // Delete file after added into app's vfs.
+                            // f.delete();
```
Comment (csadilek) [inline kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/DiagramServiceImpl.java]: delete commented out code or uncomment in case it's needed.

File: kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/DiagramServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,289 @@
...
+    }
+
+    public org.uberfire.java.nio.file.Path getDiagramsPath() {
+        return root.resolve( VFS_DIAGRAMS_PATH );
+    }
+
+    private boolean isFileNameAccepted( String name ) {
+        if ( name != null && name.trim().length() > 0 ) {
+            Set<String> exts = getExtensionsAccepted();
+            for ( String ext : exts ) {
```
Comment (csadilek) [inline kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/DiagramServiceImpl.java]: use Java8 stream anyMatch() instead of this loop.

File: kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/XMLEncoderDiagramMetadataMarshaller.java [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+import java.io.InputStream;
+
+@ApplicationScoped
+public class XMLEncoderDiagramMetadataMarshaller implements DiagramMetadataMarshaller<Metadata> {
+
+    private static final String CHARSET = "UTF-8";
+
+    @Override
+    public Metadata unmarshall( InputStream input ) throws IOException {
+        XMLDecoder decoder= new XMLDecoder( input );
```
Comment (csadilek) [inline kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/XMLEncoderDiagramMetadataMarshaller.java]: code formatting.

File: kie-wb-common-stunner/kie-wb-common-stunner-backend/src/test/java/org/kie/workbench/common/stunner/backend/service/ErraitDiagramMetadataMarshallerTest.java [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+import org.junit.Ignore;
+import org.junit.Test;
+import org.kie.workbench.common.stunner.core.diagram.Metadata;
+import org.kie.workbench.common.stunner.core.diagram.MetadataImpl;
+
+import java.io.InputStream;
+
+// TODO: @RunWith( MockitoJUnitRunner.class )
+@Ignore
+public class ErraitDiagramMetadataMarshallerTest {
```
Comment (csadilek) [inline kie-wb-common-stunner/kie-wb-common-stunner-backend/src/test/java/org/kie/workbench/common/stunner/backend/service/ErraitDiagramMetadataMarshallerTest.java]: Type in class name.

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/session/presenter/impl/AbstractClientSessionPresenter.java [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+import org.kie.workbench.common.stunner.core.client.session.impl.AbstractClientSessionManager;
+import org.kie.workbench.common.stunner.core.diagram.Diagram;
+import org.uberfire.mvp.Command;
+
+import javax.enterprise.event.Event;
+
+public abstract class AbstractClientSessionPresenter<S extends AbstractClientSession, V extends ClientSessionPresenter.View>
+        implements ClientSessionPresenter<AbstractCanvas, AbstractCanvasHandler, S, V> {
+
+    // ClientSessionPresenter<AbstractCanvas, AbstractCanvasHandler, ClientSession<AbstractCanvas, AbstractCanvasHandler>, ClientSessionPresenter.View>
```
Comment (csadilek) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/session/presenter/impl/AbstractClientSessionPresenter.java]: remove

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/toolbar/command/AbstractToolbarSessionCommand.java [hunk 14]
```diff
@@ -0,0 +1,177 @@
...
+            public void onError( final ClientRuntimeError error ) {
+                if ( null != callback ) {
+                    callback.onError( error );
+                }
+            }
+        } );
+    }
+
+    protected String getConfirmMessage() {
+        return "Are you sure?";
```
Comment (csadilek) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/toolbar/command/AbstractToolbarSessionCommand.java]: //TODO i18n , doesn't have to be now but worth noting in source.

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/toolbar/command/AbstractToolbarSessionCommand.java [hunk 15]
```diff
@@ -0,0 +1,177 @@
...
+            disable();
+        }
+
+    }
+
+
+
+
+
+
```
Comment (csadilek) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-widgets/src/main/java/org/kie/workbench/common/stunner/client/widgets/toolbar/command/AbstractToolbarSessionCommand.java]: The big void....

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-api/kie-wb-common-stunner-backend-api/src/main/java/org/kie/workbench/common/stunner/core/definition/service/DiagramMarshaller.java [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+
+package org.kie.workbench.common.stunner.core.definition.service;
+
+import org.kie.workbench.common.stunner.core.diagram.Diagram;
+import org.kie.workbench.common.stunner.core.diagram.Metadata;
+import org.kie.workbench.common.stunner.core.graph.Graph;
+
+import java.io.IOException;
+import java.io.InputStream;
+
```
Comment (csadilek) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-api/kie-wb-common-stunner-backend-api/src/main/java/org/kie/workbench/common/stunner/core/definition/service/DiagramMarshaller.java]: javadocs.

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-api/kie-wb-common-stunner-backend-api/src/main/java/org/kie/workbench/common/stunner/core/definition/service/DiagramMetadataMarshaller.java [hunk 17]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+package org.kie.workbench.common.stunner.core.definition.service;
+
+import org.kie.workbench.common.stunner.core.diagram.Metadata;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public interface DiagramMetadataMarshaller<M extends Metadata> {
```
Comment (csadilek) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-api/kie-wb-common-stunner-backend-api/src/main/java/org/kie/workbench/common/stunner/core/definition/service/DiagramMetadataMarshaller.java]: javadocs.

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-api/kie-wb-common-stunner-core-api/src/main/java/org/kie/workbench/common/stunner/core/factory/diagram/DiagramFactory.java [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+import org.kie.workbench.common.stunner.core.diagram.Metadata;
+import org.kie.workbench.common.stunner.core.factory.Factory;
+import org.kie.workbench.common.stunner.core.graph.Graph;
+
+/**
+ * Factory for Diagrams..
+ */
+public interface DiagramFactory extends Factory<Diagram, String> {
+
+    Diagram build( String name );
```
Comment (csadilek) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-api/kie-wb-common-stunner-core-api/src/main/java/org/kie/workbench/common/stunner/core/factory/diagram/DiagramFactory.java]: javaocs.

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/service/ClientDiagramServices.java [hunk 19]
```diff
@@ -17,91 +17,29 @@
...
  * A wrapper util class for handling different diagram services from client side.
  */
 @ApplicationScoped
-public class ClientDiagramServices {
-
-    SyncBeanManager beanManager;
-    DefinitionManager definitionManager;
-    Caller<DiagramService> diagramServiceCaller;
-    Caller<DiagramLookupService> diagramLookupServiceCaller;
+public class ClientDiagramServices extends AbstractClientDiagramServices<Diagram, DiagramService> {
```
Comment (csadilek) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/service/ClientDiagramServices.java]: service not services....

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/util/StunnerClientLogger.java [hunk 20]
```diff
@@ -32,4 +53,154 @@ public static String getErrorMessage( final ClientRuntimeError error ) {
...
+
+    private static void logFullSessionInfo( final AbstractClientFullSession session ) {
+        final ElementBuilderControl<AbstractCanvasHandler> builderControl = session.getBuilderControl();
+        final CanvasCommandManager<AbstractCanvasHandler> canvasCommandManager =
+                session.getCanvasCommandManager();
+        log( "Builder control = "  + ( null != builderControl ? builderControl.toString() : "null" ) );
+        log( "Canvas command mgr = "  + ( null != canvasCommandManager ? canvasCommandManager.toString() : "null" ) );
+    }
+
+        @SuppressWarnings( "unchecked" )
```
Comment (csadilek) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/util/StunnerClientLogger.java]: code formatting (please check for all files)

File: kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/AbstractDefinitionSetServices.java (hunk 1 above)
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/AbstractDefinitionSetServices.java]: agree :)

File: kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/DiagramLookupServiceImpl.java (hunk 5 above)
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/DiagramLookupServiceImpl.java]: agree :)

File: kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/AbstractDefinitionSetServices.java (hunk 2 above)
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/AbstractDefinitionSetServices.java]: not following you sorry... what you mean? about the AbstractDefinitionSetServices class?

File: kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/AbstractDiagramLookupService.java (hunk 4 above)
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/AbstractDiagramLookupService.java]: oh sure!

File: kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/DiagramServiceImpl.java (hunk 6 above)
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/DiagramServiceImpl.java]: No! There are two showcases named standalone & project ( integrates with guvnor and kie wb commons etc ). The service when uisng the standalone showcase is this Impl one, there is no project information, so it basically creates a stunner repository when starting up for first time and deploy some examples. 
On the other hand, when using the project integration artifacts, the service resolved is the concrete implementation named ProjectDiagramService... 
Basically the standalone version help us to be faster on the deverloment cycle, as the app runs twice faster, and ensures the architecture is still usable outside the workbench project environment, as Mark expects for other stuff...

File: kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/DiagramServiceImpl.java (hunk 7 above)
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/DiagramServiceImpl.java]: good point

File: kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/DiagramServiceImpl.java (hunk 8 above)
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-backend/src/main/java/org/kie/workbench/common/stunner/backend/service/DiagramServiceImpl.java]: good point

**Comment by romartin** [comment]
Thanks @csadilek ! Absolutely agree on the size of this PR, sorry.. .will take into account for further times providing smaller ones :) Agree on all changes,code formattings, etc, will provide a new commit hope in a bit. Thanks again!

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/service/ClientDiagramServices.java (hunk 19 above)
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/service/ClientDiagramServices.java]: well on client side I use "services" in plural as this class can provide shortcuts for different remote services, as currently do, it provides methods for both diagram and lookup services. But yeah, minor thing, just letting you that there was a reason hehe :) Anyway, let's be homogenious and use the singular, np! Thanks man!

**Comment by romartin** [comment]
Hi @csadilek ,
Thanks for the review! Some updates:
- Review of the different comments above
- Refactored as much as possible some collections usages to the java8 streams
- Added javadocs of some api and important interfaces and classes
- Code formatting
- Added all found i18n todos
- Rebased with master again
Thanks!!

File: kie-wb-common-stunner/kie-wb-common-stunner-backend/src/test/java/org/kie/workbench/common/stunner/backend/service/ErraitDiagramMetadataMarshallerTest.java (hunk 12 above)
Comment (csadilek) [inline kie-wb-common-stunner/kie-wb-common-stunner-backend/src/test/java/org/kie/workbench/common/stunner/backend/service/ErraitDiagramMetadataMarshallerTest.java]: @romartin want to fix the typo in the class name ErraiTDiagram or delete this class since its on ```@Ignore```?

**Comment by romartin** [comment]
Hi @csadilek , just added the commit that removes the unused test and rebased with master again. Sorry for the incoveniences. Thanks!
